### PR TITLE
Update WHD logging usage and introduce `force` flag

### PR DESCRIPTION
**Pull request by zkdev**
**What this PR does / why we need it**:
The root logger configuration is loaded, a separate handler for file output initialized and attached to the `whd` logger. Additionally a `ci.log.get_default_fmt_string()` function returning the default logging format string is introduced. Logging configuration is moved from production only to top level of WHD initialization.

Apart from the WHD logging usage the root logger config now includes the `force` parameter which indicates whether all existing handlers shall be removed. Additionally the Falcon app is now initialized using `falcon.App` as `falcon.Api` is deprecated (see: https://github.com/falconry/falcon/blob/a5e72b287efb2b3da632cf6547ed3f07d8ec5380/falcon/app.py#L1060).

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: ci/log.py [hunk 1]
```diff
@@ -44,19 +44,29 @@ class Bcolors:
...
 
     sh = logging.StreamHandler(stream=sys.stdout)
     sh.setLevel(stdout_level)
-    sh.setFormatter(CCFormatter(fmt='%(asctime)s [%(levelprefix)s] %(name)s: %(message)s'))
+    sh.setFormatter(CCFormatter(fmt=get_default_fmt_string()))
     logging.root.addHandler(hdlr=sh)
     logging.root.setLevel(level=stdout_level)
 
-    # both too verbose ...
+    # configure loggers
```
Comment (jkrayl) [inline ci/log.py:66]: I think it is better to describe why it is necessary to configure the loggers. I don't need that comment to see that the log level is set here.

**Review by jkrayl — CHANGES_REQUESTED** [review]

File: ci/log.py [hunk 2]
```diff
@@ -44,19 +44,29 @@ class Bcolors:
...
+    stdout_level=None,
+    force=False,
+):
     if not stdout_level:
         stdout_level = logging.INFO
 
     # make sure to have a clean root logger (in case setup is called multiple times)
-    logging.root.handlers.clear()
+    if force:
+        for h in logging.root.handlers[:]:
```
Comment (ccwienk) [inline ci/log.py:56]: why not simply use `handlers.clear()`? also, you can omit the slicing `[:]`
edit: ok, I suppose you want to not only rm the handlers, but also close them - correct?
still, no slicing operator required

File: ci/log.py (hunk 1 above)
Comment (ccwienk) [reply ci/log.py:66]: yes - admittedly, my previous comment was not very elaborated, but it did give some hints about the motivation for the cfg-change..

File: ci/log.py [hunk 3]
```diff
@@ -44,19 +44,29 @@ class Bcolors:
     BLUE = '\033[34m'
 
 
-def configure_default_logging(stdout_level=None):
+def configure_default_logging(
+    stdout_level=None,
+    force=False,
```
Comment (ccwienk) [inline ci/log.py:49]: is it not rather the case we do by default want the force?

File: cli/gardener_ci/whdutil.py [hunk 4]
```diff
@@ -50,6 +52,7 @@ def serve():
             p.start()
         serve()
     else:
+
```
Comment (ccwienk) [inline cli/gardener_ci/whdutil.py:55]: why?

**Review by ccwienk — CHANGES_REQUESTED** [review]
in general, I would prefer not to mix refactorings/reformattings w/ the actual "payload" (in thase case, logger-cfg)
at any rate, see inline comments. overall lgtm

File: ci/log.py (hunk 2 above)
Comment (jkrayl) [reply ci/log.py:56]: This is the way the python logging lib does it and the `logging.root.removeHandler()` method includes locking. I think you are right about the slicing tho it looks a bit redundant.

Comment (zkdev) [reply ci/log.py:56]: Yes, its about closing them as well.
Sure, will remove the slicing.

File: ci/log.py (hunk 3 above)
Comment (zkdev) [reply ci/log.py:49]: True.

**Review by jkrayl — APPROVED** [review]

**Pull request merged**